Base station Design Doc (#60)
Capstone-Skynet/Capstone-Skynet.github.io: docs/design/3_1_6_base_station.tex

One functionality that mobile devices lack is the support for common machine learning platforms (ex. TensorFlow). As the client may want to perform additional ML processing on the base station for further analysis in the future, laptops were ultimately selected as the base station's platform. Support for smart devices (to facilitate multiple viewers) may be added if time permits.

\subsubsection {Graphical User Interface}

> A web-application is used as the GUI for the base station. This allows for universal compatibility with laptops and smart mobile devices, regardless of their operating system.

Reviewer: You can probably just roll this into a the next section, this section is too short imo.

\subsubsection{Functionality}
The base station hosts a TCP/IP server, and serves two clients: the PMB and the web-application. The server will continuously query the PMB for live video data and ML metadata results, and send the data to the web-application for display.

> If an object detection scheme is implemented, the ML results will be displayed as a bounding box around the detected object(s) in the frame (\textbf{F.BS.1}). In particular, the ML results will consist of two parts: the coordinates of the upper-left vertex of the bounding box(es) and the height/width of the bounding box(es).

Reviewer: ```suggestion
The ML results will be displayed as a bounding box around the detected object(s) in the frame (\textbf{F.BS.1}). In particular, the ML results will consist of two parts: the coordinates of the upper-left vertex of the bounding box(es) and the height/width of the bounding box(es).
```

Reviewer: It's confirmed that we're doing bounding boxes

> Buttons and radio buttons are situated on the web-application for user input (\textbf{NF.BS.2}). Their functions include adjusting the visual properties of the web-application and changing settings on the PMB and PLB (\textbf{F.BS.4})

Reviewer: Which settings? We can be vague here, but that's too vague ;)

One functionality that mobile devices lack is the support for common machine learning platforms (ex. TensorFlow). As the client may want to perform additional ML processing on the base station for further analysis in the future, laptops were ultimately selected as the base station's platform. Support for smart devices (to facilitate multiple viewers) may be added if time permits.

\subsubsection{Functionality}
A web-application is used as the GUI for the base station. This allows for universal compatibility with all laptops, regardless of their operating system.

> The base station hosts a TCP/IP server, and serves two clients: the PMB and the web-application. The server contiuously queries the PMB for live video data and ML metadata results, and sends the data to the web-application for display.

Reviewer: ```suggestion
The base station hosts a TCP/IP server, and serves two clients: the PMB and the web-application. The server continuously queries the PMB for live video data and ML metadata results, and sends the data to the web-application for display.
```

The ML results are displayed as a bounding box around the detected object(s) in the frame (\textbf{F.BS.1}). In particular, the ML results consist of two parts: the coordinates of the upper-left vertex of the bounding box(es) and the height/width of the bounding box(es).

The web application also displays real-time system status messages. These messages come from the PMB or the base station itself. The status updates include information on initiation or completion of processes, changes in PMB, PLB or base station settings and system warnings and errors.

The base station continuously saves the received video and metadata locally for any analysis purposes (\textbf{F.BS.6}). All faults are also be logged locally on the device to facilitate further debugging(\textbf{F.BS.5}).
%% maybe we just log all status messages, not limited to warnings/errors.

> Buttons and radio buttons are situated on the web-application for user input (\textbf{NF.BS.2}). Their functions include adjusting the visual properties of the web-application and changing settings on the PMB and PLB (\textbf{F.BS.4}). The PMB and PLB contain adjustable settings for status message formatting, enabling and disabling, camera polling rate and ML processing enabling and disabling.

Reviewer: ```suggestion
Buttons and radio buttons are situated on the web-application for user input (\textbf{NF.BS.2}). Their functions include adjusting the visual properties of the web-application and changing settings on the PMB and PLB (\textbf{F.BS.4}). The PMB and PLB contain adjustable settings for status message formatting, camera polling rate, and ML model sensitivity.
```
